test/preprocessors/BasePreprocessor.test.ts

```diff
@@ -64,64 +74,61 @@ describe("BasePreprocessor", () => {
   });
 
   describe("#preprocess()", () => {
-    beforeEach(() => {
-      const mockBlacklistPredicate: Predicate<EntityDefinition> = {
-        evaluate: mockEvaluate,
-      };
-
-      const mockResourceTransformer: Transformer<
-        EntityDefinition,
-        EntityDefinition
-      > = {
-        transform: mockResourceTransform,
-      };
-
-      const mockElementTransformer: Transformer<
-        EntityDefinition,
-        EntityDefinition
-      > = {
-        transform: mockElementTransform,
-      };
-
-      preprocessor.blacklistPredicate = mockBlacklistPredicate;
-      preprocessor.addResourceTypeTransformer = mockResourceTransformer;
-      preprocessor.modifyElementTypeTransformer = mockElementTransformer;
-    });
-
     it("should filter out blacklisted EntityDefinitions", () => {
       const result = preprocessor.preprocess(entityCollection);
       expect(result).not.toBe(entityCollection);
-      expect(result.entities).toBeArrayOfSize(3);
-      expect(mockEvaluate).toHaveBeenCalledTimes(2);
+      expect(result.entities).toBeArrayOfSize(4);
+      expect(result.entities[0].dataType.normalizedName).toBe("Resource");
+      expect(result.entities[1].dataType.normalizedName).toBe("Element");
+      expect(result.entities[2].dataType.normalizedName).toBe("ValueSetType");
+      expect(result.entities[3].dataType.normalizedName).toBe("Type");
     });
 
     it("should add a resourceType field to Resource entities", () => {
-      preprocessor.preprocess(entityCollection);
-      expect(mockResourceTransform).toHaveBeenCalledTimes(2);
+      const result = preprocessor.preprocess(entityCollection);
+      expect(result.entities[0].dataType.normalizedName).toBe("Resource")
+      expect(result.entities[0].memberVariables).toBeArrayOfSize(3);
+      expect(result.entities[0].memberVariables[2].variableName).toBe("resourceType");
+
     });
 
     it("should modify the Element type to extend FHIR.Type", () => {
-      preprocessor.preprocess(entityCollection);
-      expect(mockElementTransform).toHaveBeenCalledTimes(2);
+      const result = preprocessor.preprocess(entityCollection);
+      expect(result.entities[1].parentDataType?.normalizedName).toBe("Type");
     });
 
     it("should create and add the FHIR.Type definition to the collection", () => {
       const result: EntityCollection = preprocessor.preprocess(
         entityCollection
       );
-      expect(result.entities).toBeArrayOfSize(3);
-      expect(result.entities[2].parentDataType).toBeNull();
-      expect(result.entities[2].dataType.namespace).toBe("FHIR");
-      expect(result.entities[2].dataType.typeName).toBe("Type");
-      expect(result.entities[2].dataType.path.toString()).toBe(
+      expect(result.entities).toBeArrayOfSize(4);
+      const fhirType = result.entities[3];
+      expect(fhirType.parentDataType).toBeNull();
+      expect(fhirType.dataType.namespace).toBe("FHIR");
```

Comment:
Codacy found an issue: [Unsafe call of an any typed value.](https://app.codacy.com/gh/MeasureAuthoringTool/model-info-parser/pullRequest?prid=5963561)